src/utils/string.hpp

```diff
@@ -459,4 +460,30 @@ inline std::string_view Substr(const std::string_view string, size_t pos = 0, si
   return string.substr(pos, len);
 }
 
+/**
+ * Convert a double value to a string representation.
+ * Precision of converted value is 16.
+ * Function also removes trailing zeros after the dot.
+ *
+ * @param value The double value to be converted.
+ *
+ * @return The string representation of the double value.
+ *
+ * @throws None
+ */
+inline std::string DoubleToString(const double value) {
+  static const int PRECISION = 15;
+  static const int STR_LENGTH = PRECISION + 1;  // 1 for dot
+
+  std::stringstream ss;
+  ss << std::setprecision(PRECISION) << std::fixed << value;
+  std::string res(ss.view(), 0, STR_LENGTH);
+
+  res.erase(res.find_last_not_of('0') + 1);
```

Comment:
Can we put a comment here saying that we are guaranteed to have at least one char different from `0` (the `.`). Otherwise we could get `npos` and we shouldn't increment it.